### ISSUE TITLE
Version 1.0.0 NPM extremely slow on compiling CSS

### ISSUE THREAD

**Issue by gleenk** [understrap/understrap#1428]
Hi guys, after migration from GULP to NPM in v 1.0.0 we noticed a drastical reduction of performance in compiling SCSS. We moved from 4s to more than 10s on the same machine. How's that possible? How to fix that? Thank you
D.

**Comment by murilodpl** [comment]
Hey I have the same problem, comparing with GULP, my NPM is 2-3 times more slower to compile... and sometimes my NPM doesn't even compile and I have to restart the "npm run watch". 
Hope we find a solution! Thank you guys

**Comment by 0dp** [comment]
For testing and comparison try and run `npm run css` and compare time against `npm run watch`

**Comment by bacoords** [comment]
Does this happen with the JS or just compiling CSS?

**Comment by gleenk** [comment]
> Does this happen with the JS or just compiling CSS?

Both


> Hey I have the same problem, comparing with GULP, my NPM is 2-3 times more slower to compile... and sometimes my NPM doesn't even compile and I have to restart the "npm run watch".

Same here, sometimes it crashes and I have to block process and restart npm run watch

**Comment by bacoords** [comment]
We'll do some more testing, though for me the JS is running 2x faster than with Gulp. 

The CSS does seem a little slower, especially the `css-compile` task. Are you modifying a parent theme or developing on a [child theme](https://github.com/understrap/understrap-child)?

**Comment by gleenk** [comment]
> We'll do some more testing, though for me the JS is running 2x faster than with Gulp.
> 
> The CSS does seem a little slower, especially the `css-compile` task. Are you modifying a parent theme or developing on a [child theme](https://github.com/understrap/understrap-child)?

I'm working on child theme

**Comment by verybg** [comment]
There is extremely slow on compiling CSS same for me.

**Comment by 0dp** [comment]
It's very hard to debug this or come with some good answers when we don't have any hard data from you guys.

Libsass was written in C/C++ with compiled binaries which will always be faster than dart sass, which i believe is compiled to JavaScript. There is an ongoing effort to make it faster as a js lib. But i do believe that it's possible to run sass via their dart VM which should be fast fast fast.

But i have no proof or justification for above statement really. If someone wants to try and compile the sass with sass CLI and compare maybe that'll confirm that theory.

But then we are also saying that js is slow. This throws me off a bit. Then it has nothing to do with sass or js but should be a rollup issue?

**Comment by gleenk** [comment]
I did tests with "vanilla" version of undestrap, with new project from scratch only installing main theme and child and running npm. nothing, nothing else. I know that's not really useful but i'm not an engineer, only a frontend developer :D

**Comment by bacoords** [comment]
As @0dp is saying, previous versions that used `gulp-sass` were relying on `node-sass` (LibSass), but this tool is now deprecated. Technically you can keep using it, and there's no reason you can't start a child theme from [0.9.6](https://github.com/understrap/understrap-child/releases) and use the old gulp build process. We could even add that messaging to the readme. Or you can even try to replace `sass` with `node-sass` in the `package.json`, which should still be a one-to-one swap as far as the commands go.

This is also an unfortunate side-effect of the javascriptification of all things, but if the choice is between using a deprecated tool and following the lead of Bootstrap (who's build process we're trying to follow closely), this is where we're at. We're working on [Bootstrap 5 integration](https://github.com/understrap/understrap/tree/feature/bs5-development) as well, so that also plays into what tools we can use going forward.

However, maybe there are some config options we could look into to be more streamlined and speed up the actual compilation. Maybe someone could do a pure postCSS way to compile and remove `sass` entirely. It's also hard due to how Bootstrap makes heavy use of mixins.

As for JS, I haven't replicated the speed differences, but maybe I'll pull it up on a different computer and see if that effects it. We've gone from just uglifying to doing module bundling, so there's a little more going on (but also the ability to be more lean in your final JS files).

**Comment by gleenk** [comment]
Just to answer, what's the main hardware involved in speeding up NPM / SASS compilation? Processor or ram?

**Comment by schwarzpunkt** [comment]
As long as you don't need, you can delete "css-postcss" in the script "css" in package.json. npm run css will be a little faster then. I prefer to manage editor-color-palette.json myself anyway.

**Comment by 0dp** [comment]
I wouldn't circumvent postcss since it handles the autoprefixer. You might run in to compatibility issues in on other browsers/devices.

What you could try is maybe is set annotation and sourcesContent to false.

@bacoords  i don't know, but i think it's a bit sus that we are doing some sourcemap processing in postcss when also doing it directly in build scripts commands. Yai/nai?

**Comment by bacoords** [comment]
Yeah postcss also handles prefixing, and that task seems to be pretty quick to me. It's the compile task that seems to take the longest.

Agreed - I'll make a note to review the source maps. It feels like you typically have source maps for the .css but not the .min.css, correct? I know bootstrap includes them for both.

**Comment by 0dp** [comment]
I think they include it for both because their site also works as documentation.

Personally, I'd avoid having the extra KB in production.

**Comment by bacoords** [comment]
We are working on this - please check this issue: https://github.com/understrap/understrap/issues/1598 for more info about `sass-embedded` and let us know if you can test it out and see if it improves your speed.
